lib/Varien/Db/Tree/NodeSet.php

```diff
@@ -58,13 +58,10 @@ public function valid(): bool
         return isset($this->_nodes[$this->_current]);
     }
 
-    #[\ReturnTypeWillChange]
     public function next()
     {
-        if ($this->_current > $this->_currentNode) {
-            return false;
-        } else {
-            return $this->_current++;
+        if ($this->_current <= $this->_currentNode) {
+            $this->_current++;
```

Comment:
Nothing returned?

Reply:
> Nothing returned?

Per the `\Iterator` interface, `next()` should only advance the pointer forward.


Have you checked if any code relied on the return type?

Reply:
I did not found any code that directly uses this, but there may be 3rd party code that relies on this.

This changes the "returned" value to `null` for no reason. Isnt `#[\ReturnTypeWillChange]` excactly for this?

Reply:
it doesn't seem the best to implement an interface and then do whatever you want, but anyway phpstan is complaining that that method shouldn't return anything.